pkg/controllers/namespace/namespace_sync_controller.go

```diff
@@ -110,21 +129,38 @@ func (c *Controller) buildWorks(namespace *corev1.Namespace, clusters []clusterv
 			OwnerReferences: []metav1.OwnerReference{
 				*metav1.NewControllerRef(namespace, namespace.GroupVersionKind()),
 			},
+			Annotations: annotations,
 		}
 
-		util.MergeLabel(namespaceObj, workv1alpha1.WorkNamespaceLabel, workNamespace)
-		util.MergeLabel(namespaceObj, workv1alpha1.WorkNameLabel, workName)
+		util.MergeLabel(clonedNamespaced, workv1alpha1.WorkNamespaceLabel, workNamespace)
+		util.MergeLabel(clonedNamespaced, workv1alpha1.WorkNameLabel, workName)
 
-		if err = helper.CreateOrUpdateWork(c.Client, objectMeta, namespaceObj); err != nil {
+		if err = helper.CreateOrUpdateWork(c.Client, objectMeta, clonedNamespaced); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func recordAppliedOverrides(cops *overridemanager.AppliedOverrides) (map[string]string, error) {
```

Comment:
This function is just same as `recordAppliedOverrides` in pkg/controllers/binding/common.go:21, we can make it an exported method in pkg/util/annotation.go.